backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java

```diff
@@ -8,49 +8,48 @@
-  public List<TestEvent> queryMatchAllBetweenDates(Date before, Date after, Pageable p);
-
-  @Query(
-      value =
-          "WITH FILTEREDEVENTS AS ("
-              + " SELECT DISTINCT ON (test_order_id) * "
-              + " FROM {h-schema}test_event te "
-              + " WHERE te.facility_id = :facilityId "
-              + " ORDER BY test_order_id, te.created_at desc"
-              + ") "
-              + " SELECT * FROM FILTEREDEVENTS "
-              + " WHERE created_at > :newerThanDate "
-              + // moving this filter into the CTE makes this query significantly
-              // more efficient (like 75% faster in one case), but then when we
-              // make it more complicated somebody will probably break it
-              " ORDER BY created_at DESC ",
-      nativeQuery = true)
-  public List<TestEvent> getTestEventResults(UUID facilityId, Date newerThanDate);
-
-  //	@Query("FROM #{#entityName} q WHERE q.facility = :facility and q.createdAt > :newerThanDate
-  // ORDER BY q.createdAt DESC")
-  //	@EntityGraph(attributePaths = {"patient", "order", "order.patientLink"})
-  //	public List<TestEvent> getTestEventResults(Facility facility, Date newerThanDate);
+    @Query("FROM #{#entityName} e WHERE e.patient = :p and e.facility in :facilities")
+    public List<TestEvent> findAllByPatientAndFacilities(Person p, Collection<Facility> facilities);
+
+    public List<TestEvent> findAllByOrganizationOrderByCreatedAtDesc(Organization o);
+
+    public List<TestEvent> findAllByOrganizationAndFacility(Organization o, Facility f);
+
+    public TestEvent findFirst1ByPatientOrderByCreatedAtDesc(Person p);
+
+    @EntityGraph(attributePaths = { "patient", "order" })
+    public TestEvent findByOrganizationAndInternalId(Organization o, UUID id);
+
+    // Need to control how this query is built. "between" is too vague.
+    // This is across all Orgs/facilities because datahub uploader users
+    @Query("FROM #{#entityName} q WHERE q.createdAt > :before AND q.createdAt <= :after ORDER BY q.createdAt")
+    public List<TestEvent> queryMatchAllBetweenDates(Date before, Date after, Pageable p);
+
+    @Query(value = "WITH FILTEREDEVENTS AS (" + " SELECT DISTINCT ON (test_order_id) * "
+            + " FROM {h-schema}test_event te " + " WHERE te.facility_id = :facilityId "
+            + " ORDER BY test_order_id, te.created_at desc" + ") " + " SELECT * FROM FILTEREDEVENTS "
+            // moving this filter into the CTE makes this query significantly
+            // more efficient (like 75% faster in one case), but then when we
+            // make it more complicated somebody will probably break it
+            + " ORDER BY created_at DESC ", countQuery = "SELECT count(*) FROM FILTEREDEVENTS", nativeQuery = true)
+    public List<TestEvent> getTestEventResults(UUID facilityId, Pageable pageable);
+
+    @Query(value = "WITH FILTEREDEVENTS AS (" + " SELECT DISTINCT ON (test_order_id) * "
```

Comment:
Is this not optimal? Really i just want a count of the results specified in the query above